src/Model/IndexInfo.php

```diff
@@ -70,6 +74,43 @@ public function __toString()
         return $this->getName();
     }
 
+    /**
+     * Return the 2dsphere index version number.
+     *
+     * @return integer|null
+     */
+    public function get2dSphereIndexVersion()
+    {
+        if ( ! empty($this->info['2dsphereIndexVersion'])) {
+            return $this->info['2dsphereIndexVersion'];
+        }
+    }
+
+    /**
+     * Return the bucketSize for geoHaystack indexes.
+     *
+     * @return double|null
+     */
+    public function getBucketSize()
+    {
+        if ( ! empty($this->info['bucketSize'])) {
+            return $this->info['bucketSize'];
+        }
+    }
+
+    /**
+     * Return the default language of a text index.
+     *
+     * @return string
+     */
+    public function getDefaultLanguage()
+    {
+        if (empty($this->info['default_language'])) {
+            return self::DEFAULT_LANGUAGE;
```

Comment:
I'm quite quite sure whether we should return the defaults ``self::DEFAULT_LANGUAGE_OVERRIDE`` and ``self::DEFAULT_LANGUAGE`` that you have defined, or just ``null``, as all non-text indexes will not have this information, and returning ``english``/``language`` might not be expected here.